docs/design/3_1_4_machine_learning.tex

```diff
@@ -1 +1,29 @@
-TODO: Describe potential machine learning applications. What could this look like, what will this \textit{not} look like? What kind of accelerators already exist? What might our solution look like?
\ No newline at end of file
+% TODO: Describe potential machine learning applications. What could this look like, what will this \textit{not} look like? What kind of accelerators already exist? What might our solution look like?
+
+% 
+
+
+Machine learning is the topic that provides the context of the project. This project is not focused on creating a machine learning model, but creating a mobile computing platform that will allow the client to perform research in machine-learning models.
+
+A mobile computing platform means we are able to perform processing in unaccessilble locations. Using a drone, we are creating an unmanned aerial vehicle with the ability for its user to change operating functions using programmable logic.
+
+A user can program the logic to perform one application, allow the drone to perform that function autonomously, and then program the logic for a completely differnent application, without having switch its processing hardware.
+
+A computing platform with machine learning means that it can automatically learn and improve from experiences without being explicitly programmed. This concept is accentuated when the computing platform becomes mobile; the system can autonomously serve its purpose without needing physically close monitoring.
+
+\subsubsection{Correct Machine Learning Applications}
+Because this is a mobile platform, the computing platform serves to take in physical data from the external world, process it, and output data that can be analyzed. Examples of this usage would include image/video recognition, as well as pattern recognition and classification from an aerial environment.
+
+\subsubsection{Incorrect Machine Learning Applications}
+The computing platform would not serve its purpose in applications where the data to be processed is sent by its user. The mobile computing platform should not be used in applications such as email classification, spam filtering and virtual assistants.
+
+Also factoring in the mobility of the system, the machine learning implementation should not be used in applications should be performed on the ground, even if ML can be applied. Cases of grounded applications include medical services and speech recognition.
+
+% \subsubsection{Hardware Accelerators} %????
+% The hardware accelerators used for machine learning can be applied in CPUs, GPUs, FPGAs and ASICs. For this project we are focusing on FPGA hardware accelerated machine learning. 
+
+\subsubsection{Project Implementation}
```

Comment:
Just a general comment - I don't know how much of the above should be in a design document. I think I should've thought about my TODO comment more, my bad. 

I know the machine learning portion is still vague right now, so I think we can cut a lot of this (it's okay if this section is smaller than the others). I don't think we need to discuss things like tangential ML applications like virtual assistants. I also don't think we need to discuss what it means to do ML in general.

What I'd suggest is that we include the first blurb ("Machine learning is the topic that provides the context of the project..."), the last blurb ("Our solution would be a simple hardware accelerated machine learning algorithm..."), and a description of possible implementations (pedestrian detection, cattle ranching, etc, as written in the proposal). We should also reference the FPGA section here, as we discuss the ML acceleration more there.